### PR TITLE
Removing more miscellaneous throw specifiers.

### PR DESCRIPTION
**Pull request by timothy-king**
Also fixing the spelling of amount in several places.

### CODE REVIEW

File: src/theory/bv/theory_bv_rewrite_rules_normalization.h [hunk 1]
```diff
@@ -529,15 +529,15 @@ bool RewriteRule<ConcatToMult>::applies(TNode node) {
...
   TNode extract = node[0];
   TNode c = node[1];
-  unsigned ammount = utils::getSize(c);
-  
-  if (utils::getSize(node) != utils::getSize(extract[0])) return false; 
-  if (c != utils::mkConst(ammount, 0)) return false;
+  unsigned amount = utils::getSize(c);
+
+  if (utils::getSize(node) != utils::getSize(extract[0])) return false;
+  if (c != utils::mkConst(amount, 0)) return false;
```
Comment (aniemetz) [inline src/theory/bv/theory_bv_rewrite_rules_normalization.h]: Use `utils::mkZero(amount)`

File: src/theory/bv/theory_bv_rewrite_rules_simplification.h [hunk 2]
```diff
@@ -1127,28 +1127,34 @@ bool RewriteRule<MergeSignExtend>::applies(TNode node) {
...
     return res;
   }
   Assert (node[0].getKind() == kind::BITVECTOR_SIGN_EXTEND);
-  unsigned ammount2 = node[0].getOperator().getConst<BitVectorSignExtend>().signExtendAmount;
+  unsigned amount2 =
+      node[0].getOperator().getConst<BitVectorSignExtend>().signExtendAmount;
   NodeBuilder<> nb(kind::BITVECTOR_SIGN_EXTEND);
-  Node op = nm->mkConst<BitVectorSignExtend>(BitVectorSignExtend(ammount1+ ammount2));
+  Node op =
+      nm->mkConst<BitVectorSignExtend>(BitVectorSignExtend(amount1 + amount2));
```
Comment (aniemetz) [inline src/theory/bv/theory_bv_rewrite_rules_simplification.h]: Use `utils::mkSignExtend`? Might be easier to read than using a NodeBuilder.

Comment (timothy-king) [reply src/theory/bv/theory_bv_rewrite_rules_simplification.h]: done

File: src/theory/bv/theory_bv_rewrite_rules_normalization.h (hunk 1 above)
Comment (timothy-king) [reply src/theory/bv/theory_bv_rewrite_rules_normalization.h]: done

**Pull request merged**